### PR TITLE
testcase/filesystem : Test more negative cases of setvbuf()

### PR DESCRIPTION
**Pull request by hasw7569**
testcase/filesystem : Test more negative cases of setvbuf() 

1. Fail to file open
2. Invalid mode check
3. If a buffer pointer is provided, then it must have a non-zero size
4. Reuse buffer
5. EBUSY test

IMXRT1050-EVKB log :

[tc_libc_stdio_setvbuf] PASS

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [bbc06e1f074fd6318a242baffe4b39d63d3b46e8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [bbc06e1f074fd6318a242baffe4b39d63d3b46e8] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 1]
```diff
@@ -2441,16 +2441,51 @@ static void tc_libc_stdio_setbuf(void)
...
+	fp_fd = fp->fs_fd;
+	fp->fs_fd = ERROR;
+	ret = setvbuf(fp, buffer, _IOFBF, 64);
+	fp->fs_fd = fp_fd;
+	TC_ASSERT_LEQ_CLEANUP("setvbuf", ret, 0, fclose(fp));
+	TC_ASSERT_EQ_CLEANUP("setvbuf", *get_errno_ptr(), EBADF, fclose(fp));
+
+	/* setvbuf_test: Invalid mode check */
+
+	ret = setvbuf(fp, NULL, -1, 0);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: Please use defined value for readability about invalid mode.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 2]
```diff
@@ -2441,16 +2441,51 @@ static void tc_libc_stdio_setbuf(void)
...
+
+	/* setvbuf_test: If a buffer pointer is provided, then it must have a non-zero size */
+
+	ret = setvbuf(fp, buffer, _IOFBF, 0);
+	TC_ASSERT_LEQ_CLEANUP("setvbuf", ret, 0, fclose(fp));
+	TC_ASSERT_EQ_CLEANUP("setvbuf", *get_errno_ptr(), EINVAL, fclose(fp));
+
+	/* setvbuf_test: Reuse buffer */
+
+	ret = setvbuf(fp, NULL, _IOFBF, 0);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: In this case, there is a memory allocation for buffer.
Is it okay not to free the allocated buffer after test?

**Comment by seinfra** [comment]
Target : [83f2698a9a50dd930b518258169aa41ce5c5adff] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [83f2698a9a50dd930b518258169aa41ce5c5adff] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 3]
```diff
@@ -2441,16 +2441,51 @@ static void tc_libc_stdio_setbuf(void)
 #if CONFIG_STDIO_BUFFER_SIZE > 0
 static void tc_libc_stdio_setvbuf(void)
 {
-	FILE *fp;
+	FILE *fp = NULL;
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: Why is this necessary?

**Comment by seinfra** [comment]
Target : [9d948159eac99bf4574fe2fc09d439c0ad488825] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9d948159eac99bf4574fe2fc09d439c0ad488825] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 1 above)
Comment (hasw7569) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: modify

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 3 above)
Comment (hasw7569) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: It is not necessary.
modify

**Comment by sunghan-chang** [comment]
@hasw7569 This is not an adding new testcase. To clarify, let's rewrite the title.
```
testcase/filesystem: test more negative cases of setvbuf()

1. fail to file open
2. Invalid mode check
3. If a buffer pointer is provided, then it must have a non-zero size
4. Reuse buffer
5. EBUSY test
```

**Comment by seinfra** [comment]
Target : [363563d50c405816053412940b1040da05e2f88c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [363563d50c405816053412940b1040da05e2f88c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3bbd860c88615bfcabd50db80afc20d620bb4921] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3bbd860c88615bfcabd50db80afc20d620bb4921] - Code Rule Check (C++) OK.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 4]
```diff
@@ -2445,12 +2445,51 @@ static void tc_libc_stdio_setvbuf(void)
...
 	TC_ASSERT_NEQ("fopen", fp, NULL);
 
+	/* setvbuf_test: fail to file open */
+
+	fp_fd = fp->fs_fd;
+	fp->fs_fd = ERROR;
+	ret = setvbuf(fp, buffer, _IOFBF, 64);
+	fp->fs_fd = fp_fd;
+	TC_ASSERT_LEQ_CLEANUP("setvbuf", ret, 0, fclose(fp));
+	TC_ASSERT_EQ_CLEANUP("setvbuf", *get_errno_ptr(), EBADF, fclose(fp));
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: You should use `errno` instead of `*get_errno_ptr()` for protected build.

Comment (hasw7569) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: modify

**Comment by seinfra** [comment]
Target : [c19167f1ae93c85fbd241064e4c3c175897ce1dc] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c19167f1ae93c85fbd241064e4c3c175897ce1dc] - Code Rule Check (C++) OK.
